blog: add training on any cloud with Tigris post (#175)
tigrisdata/tigris-blog: blog/2024-11-22-training-any-cloud/index.mdx

    Iaso.
  </details>
</div>

When you get started with finetuning AI models, you typically pull the datasets
and models from somewhere like the Hugging Face Hub. This is generally fine, but
as your usecase grows and gets more complicated, you're going to run into two
big risks:

- You're going to depend on the things that are critical to your business being
  hosted by someone else on a platform that doesn't have a public SLA
  (Service-Level Agreement, or commitment to uptime with financial penalties
  when it is violated).
- Your dataset will grow beyond what you can fit into ram (or even your hard
  disk), and you'll have to start sharding it into chunks that are smaller than
  ram.

Most of the stuff you'll find online deals with the "happy path" of training AI
models, but the real world is not quite as kind as this happy path is. Your data
will be bigger than ram. You will end up needing to make your own copies of
datasets and models because they will be taken offline without warning. You will
need to be able to move your work between providers because price hikes will
happen.

The unfortunate part is that this is the place where you're left to figure it

> out on your own. Let's break down how to do this in the real world with a flow

Reviewer: ```suggestion
out on your own. Let's break down how to build model training in the real world with a flow
```

that can expand to any dataset, model, or cloud provider with minimal changes
required. We're going to show you how to use Tigris to store your datasets and
models, and how to use
[SkyPilot](https://skypilot.readthedocs.io/en/latest/docs/index.html) to
abstract away the compute layer so that you can focus on the actual work of
training models. This will help you reduce the risk involved with training AI
models on custom datasets by importing those datasets and models once, and then
always using that copy for training and inference.

{/* truncate */}

:::note

At the highest level, this flow is optimized for AI training, but the
metapattern I spell out here is also applicable to any other environment where
you can spin up compute with just an API call. If you're not training AI models,
you can still use the general metapattern to make your multi-cloud life easier
by storing your state or needed data files in Tigris.

:::

## Non-fungible compute

It’s kinda remarkable that we’ve ended up in a situation where compute is so
fungible, but in order to get to the point that you can do anything with it you

> need to introduce so much undefined state and just hope that things work out.

Reviewer: ```suggestion
need to introduce a bunch of undefined state and just hope that things work out.
```
